### PR TITLE
mkvtoolnix only mojave or later

### PR DESCRIPTION
**Pull request by dericed**
Related to https://github.com/amiaopensource/vrecord/issues/577 & https://github.com/amiaopensource/homebrew-amiaos/issues/258.

@bturkus, @tomackroyd, please test.

### CODE REVIEW

File: vrecord.rb [hunk 1]
```diff
@@ -35,6 +36,20 @@ def caveats
...
+  if MacOS.version < :mojave
+    def caveats
+      <<~EOS
+        ** IMPORTANT FOR macOS INSTALL **
+        Additional install steps are necessary for a fully functioning Vrecord
+        install on macOS. Vrecord uses mkvtoolnix for some optional steps such
+        as embedding logs into Matroska files. mkvtoolnix is automatically
+        installed via homebrew if the Mac OS is Mojave or greater, else we
+        recommend that mkvtoolnix is installed via
+        https://mkvtoolnix.download/macos/MKVToolNix-46.0.0.dmg
```
Comment (retokromer) [inline vrecord.rb:48]: Is there any reason to choose 46? The current one is 49. I would prefer a more generic link, such as `https://mkvtoolnix.download/macos/`, indicating to chose the last available release.

File: vrecord.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@ class Vrecord < Formula
   version "2020-07-01"
   sha256 "983264ca6a69b78b4487a7479ab5a4db04cbc425f865ec2cb15844e72af4f4ac"
   head "https://github.com/amiaopensource/vrecord.git"
+  revision 1
```
Comment (retokromer) [inline vrecord.rb:8]: Please put the `revision` line before the `head` line.

**Comment by tomackroyd** [comment]
So, I'm just running
`brew tap amiaopensource/amiaos`
`brew upgrade vrecord`?

**Comment by retokromer** [comment]
You need to run locally the [amended formula](https://raw.githubusercontent.com/amiaopensource/homebrew-amiaos/4001b8541444edbdd9b7187a6a2a46c9a435d4cf/vrecord.rb). I guess, the easiest is to save it on your computer and then drag-and-drop into the Terminal:
```
brew uninstall vrecord
brew install <drag&drop the amended formula>
```

**Comment by tomackroyd** [comment]
> You need to run locally the [amended formula](https://raw.githubusercontent.com/amiaopensource/homebrew-amiaos/4001b8541444edbdd9b7187a6a2a46c9a435d4cf/vrecord.rb). I guess, the easiest is to save it on your computer and then drag-and-drop into the Terminal:
> 
> ```
> brew uninstall vrecord
> brew install <drag&drop the amended formula>
> ```

```Last login: Thu Aug  6 19:02:55 on ttys000
GPT1:~ GPT1$ /Users/GPT1/Desktop/vrecord.rb
-bash: /Users/GPT1/Desktop/vrecord.rb: Permission denied
GPT1:~ GPT1$```

**Comment by retokromer** [comment]
The formula has to be run with `brew install` or `brew upgrade` not as a standalone Bash command.

**Comment by retokromer** [comment]
```
brew install /Users/GPT1/Desktop/vrecord.rb
```
in your case.

**Comment by tomackroyd** [comment]
This install failed installing a dependency - icu4c -  probably unrelated. Log:
[amiaos259_vrecord_brew_install_log_2020-08-06.txt](https://github.com/amiaopensource/homebrew-amiaos/files/5034231/amiaos259_vrecord_brew_install_log_2020-08-06.txt)
And here's a zip of ~/Library/Logs/Homebrew/icu4c which might help 
[icu4c.zip](https://github.com/amiaopensource/homebrew-amiaos/files/5034237/icu4c.zip)
Tom

**Comment by retokromer** [comment]
Please try:
```
brew install -cc=gcc-10 icu4c
```
and install only after this vrecord.

**Comment by dericed** [comment]
Hi @tomackroyd, yikes, an icu4c issue is hard to workaround as it's a dependency of harfbuzz which is a dependency of pango (dependency of gnuplot), and libass (dependency of ffmpeg which is a dependency of mpv, and ffmpeg (dependency of qcli). Since qcli, gnuplot, and mpv are all now optional dependencies, vrecord could still run but just with related features suppressed. I suggest seeing if there's a workaround for the icu4c issues first, if not, we could do the same with those dependencies as we did with mkvtoolnix.

**Comment by dericed** [comment]
Hi @tomackroyd, note that this PR was just updated to be based upon vrecord release 2020-08-05. If the icu4c part can't be worked around, I made a draft PR at https://github.com/amiaopensource/homebrew-amiaos/pull/261 which removes the need for it in the vrecord dependency tree, (`brew deps --tree vrecord`).

**Comment by retokromer** [comment]
On my end, `brew install -cc=gcc-10 icu4c` works on macOS 10.12 and vrecord runs fine.

**Comment by bturkus** [comment]
This seemed to work my end (I got the July version of vrecord installed on a High Sierra machine), but I did have a few issues:
-  mpv didn't install properly (but this felt like an older and potentially ongoing issue)
-  dv capture wasn't recognizing my ddv device, and I had to install avfctl separately to get it going.

I'm not sure if either of these are truly related to this fix, but wanted to report nonetheless. Seriously, thanks for working on this!

**Comment by retokromer** [comment]
@bturkus On my end, I didn’t check DV, but I will do it. What is the problem with `mpv` you encountered? Here it works fine and the issue should be fixed upstream since a while.

File: vrecord.rb (hunk 1 above)
Comment (privatezero) [reply vrecord.rb:48]: I note that at https://mkvtoolnix.download/downloads.html#macosx it says

> Up to and including MKVToolNix v41, the DMGs worked on at least 10.12 "Sierra" and newer. 

Does anything newer than v41 work on 10.12 - or would it be worth pointing to that version?

Comment (retokromer) [reply vrecord.rb:49]: >Does anything newer than v41 work on 10.12 - or would it be worth pointing to that version?

Hmm… I didn’t notice that. Thank you for the information! Do you have any clue why? The “C++17 issue”, which appeared in version 47, can be resolved with a small patch, as I pointed out earlier.

**Comment by tomackroyd** [comment]
> On my end, `brew install -cc=gcc-10 icu4c` works on macOS 10.12 and vrecord runs fine.

I ran this and after ```==> Upgrading icu4c 64.2 -> 67.1 ``` process fails with similar error.

**Comment by retokromer** [comment]
@tomackroyd Thank you for reporting back!

Could you possibly run
```
which -a gcc
```
and then version on the elements that show up, for example:
```
/usr/local/bin/gcc --version
/usr/bin/gcc --version
```

File: vrecord.rb (hunk 1 above)
Comment (privatezero) [reply vrecord.rb:49]: Absolutely no idea why! (Or if that is even accurate). I have no way to test unfortunately

Comment (retokromer) [reply vrecord.rb:49]: No problem. I’ll try to find the time this weekend to play a little with old compilers… usually they work best with old software.

**Comment by tomackroyd** [comment]
> @tomackroyd Thank you for reporting back!

Thanks @retokromer, here is the result:
```GPT1:~ GPT1$ /usr/bin/gcc --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Comment by retokromer** [comment]
Sorry, I’m lost! @tomackroyd There is a `gcc-10` and right now I have no clue at all why this is not compiling `icu4c` on your end.

**Comment by dericed** [comment]
Updated the pr to workaround any need for icu4c installation, so these dependencies are dropped from install if less than mojave.

```
amiaopensource/amiaos/vrecord
├── gnuplot
│   ├── pango
│   │   └── harfbuzz
│   │       └── icu4c
├── mkvtoolnix
│   ├── boost
│   │   └── icu4c
├── mpv
│   ├── ffmpeg
│   │   ├── libass
│   │   │   └── harfbuzz
│   │   │       └── icu4c
├── qcli
│   ├── ffmpeg
│   │   ├── libass
│   │   │   └── harfbuzz
│   │   │       └── icu4c
```

Please re-test.

**Comment by tomackroyd** [comment]
Will attempt this in the next few days. I just need to brew install using the updated .rb, right @dericed?

**Comment by dericed** [comment]
Hi @tomackroyd, yep, the one at https://raw.githubusercontent.com/amiaopensource/homebrew-amiaos/d1626a70d2d13b387f8ac9e674f9836caf38541e/vrecord.rb. Or `brew upgrade https://raw.githubusercontent.com/amiaopensource/homebrew-amiaos/d1626a70d2d13b387f8ac9e674f9836caf38541e/vrecord.rb` may work. @bturkus, could you test this as well?

**Comment by tomackroyd** [comment]
Hello all. All seemed to be going well but this process ended up running ffmpeg dvrescue/dvloupe tests and terminal has now stalled with 
[Terminal Saved Output.txt](https://github.com/amiaopensource/homebrew-amiaos/files/5061047/Terminal.Saved.Output.txt)

**Comment by dericed** [comment]
Hi @tomackroyd, hmm, that error would be from using an older version of bash. Try `brew install bash`, though the installer should do that.

**Comment by retokromer** [comment]
Indeed, Apple computer still come with a Bash 3.2 (they pretend for legal reasons, no joke).

**Comment by retokromer** [comment]
>Try `brew install bash`, though the installer should do that.

@dericed If I remember carefully, by default it’s still Apple’s one that is running.

**Comment by retokromer** [comment]
@tomackroyd Please check if you have more that one Bash installed: `cat /etc/shells`.

**Comment by tomackroyd** [comment]
> @tomackroyd Please check if you have more that one Bash installed: `which -a bash` and `cat /etc/shells`.

An upgrade or install I ran recently (and I didn't keep logs) upgraded bash to 5.0.18 - so yes I have two versions and am now hopelessly lost.
There is /usr/local/bin/bash which is an alias to /usr/local/Cellar/bash/5.0.18/bin
and /bin/bash which I'm guessing is 3.2
[bash_audit_2020-08-15.txt](https://github.com/amiaopensource/homebrew-amiaos/files/5078085/bash_audit_2020-08-15.txt)

**Comment by tomackroyd** [comment]
> > Try `brew install bash`, though the installer should do that.
> 
> @dericed If I remember carefully, by default it’s still Apple’s one that is running.

I think this is true, see above.

**Comment by retokromer** [comment]
Both are installed (I mean: Apple’s historic flavour and Homebrew’s current one), but the newer is not in your path. Different ways do change this are described in Daniel Weibel: [Upgrading Bash on macOS](https://itnext.io/upgrading-bash-on-macos-7138bd1066ba).

The easiest is probably to add at the beginning of the list `/usr/local/bin/bash` (and, optionally, to “comment out” the `/bin/bash` line):
```
# List of acceptable shells for chpass(1).
# Ftpd will not allow users to connect who are not using
# one of these shells.

/usr/local/bin/bash
#/bin/bash
/bin/csh
/bin/ksh
/bin/sh
/bin/tcsh
/bin/zsh
```

I personally prefer a more radical solution: `chsh -s /usr/local/bin/bash`, but that’s a matter of taste.

**Comment by retokromer** [comment]
@tomackroyd I forgot, you can check with `bash -c 'echo $BASH_VERSION'` which version is actually running, when multiple are installed.
@dericed We could add the Bash version check in the script. (I guess, we need 4.0 or 4.2, but I haven’t verified it.)

**Comment by tomackroyd** [comment]
> @tomackroyd I forgot, you can check with `bash -c 'echo $BASH_VERSION'` which version is actually running, when multiple are installed.
> @dericed We could add the Bash version check in the script. (I guess, we need 4.0 or 4.2, but I haven’t verified it.)

I have followed Daniel Weibel's process. Still flummoxed. Do I need to have the new bash as default or Apple's preferred?
```
Last login: Sat Aug 15 16:53:42 on ttys000
GPT1:~ GPT1$ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin16)
Copyright (C) 2007 Free Software Foundation, Inc.
GPT1:~ GPT1$ cat /etc/shells
# List of acceptable shells for chpass(1).
# Ftpd will not allow users to connect who are not using
# one of these shells.

/usr/local/bin/bash
#/bin/bash
/bin/csh
/bin/ksh
/bin/sh
/bin/tcsh
/bin/zsh
GPT1:~ GPT1$ sudo chsh -s /usr/local/bin/bash
Password:
Changing shell for root.
GPT1:~ GPT1$ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin16)
Copyright (C) 2007 Free Software Foundation, Inc.
GPT1:~ GPT1$ echo $BASH_VERSION
5.0.18(1)-release
GPT1:~ GPT1$ #!/bin/bash
GPT1:~ GPT1$ echo $BASH_VERSION
5.0.18(1)-release
GPT1:~ GPT1$ #!/usr/bin/env bash
GPT1:~ GPT1$ echo $BASH_VERSION
5.0.18(1)-release
GPT1:~ GPT1$ brew install /Users/GPT1/Desktop/vrecord.rb
Updating Homebrew...
Warning: vrecord 2020-08-05_1 is already installed and up-to-date
To reinstall 2020-08-05_1, run `brew reinstall vrecord`
GPT1:~ GPT1$ vrecord -g
/usr/local/bin/vrecord: line 390: syntax error near unexpected token `;;'
/usr/local/bin/vrecord: line 390: `            "FFV1 version 3")   SUFFIX="_ffv1" ;;'
GPT1:~ GPT1$ bash -c 'echo $BASH_VERSION'
3.2.57(1)-release
GPT1:~ GPT1$
```

**Comment by retokromer** [comment]
Have you closed and re-opened the Terminal? (Otherwise the new path is not “active”.)

**Comment by tomackroyd** [comment]
> Have you closed and re-opened the Terminal? (Otherwise the new path is not “active”.)

Have done several times and have restarted machine. 
```
GPT1:~ GPT1$ bash -c 'echo $BASH_VERSION'
3.2.57(1)-release
GPT1:~ GPT1$ sudo chsh -s /usr/local/bin/bash
Password:
Changing shell for root.
chsh: no changes made
GPT1:~ GPT1$ 
```
(restart terminal)
```
GPT1:~ GPT1$ bash -c 'echo $BASH_VERSION'
3.2.57(1)-release
GPT1:~ GPT1$ echo $BASH_VERSION
5.0.18(1)-release
GPT1:~ GPT1$ 

```

**Comment by retokromer** [comment]
Hmm… Try `sudo chsh -s /usr/local/bin/bash` (this sets the default also at administrator level).

**Comment by tomackroyd** [comment]
> Hmm… Try `sudo chsh -s /usr/local/bin/bash` (this sets the default also at administrator level).

I already did that - see my last post - did I miss something? (And thanks so much for all your help, Reto.)

**Comment by retokromer** [comment]
Sorry, at the moment, I have no clue. That’s what we have done here on the few remaining macOS computers. I will try to find a better later today (now I am running to the market, before the rush hours :-)

**Comment by retokromer** [comment]
FYI: [WIP](https://github.com/unicode-org/icu/commit/715d254a#diff-470d8af30594f47058b7e31baf227763) on Homebrew to fix `icu4c` for macOS from 10.10 to 10.12.

**Pull request merged**